src/static/js/modules/jquery/cf_formValidator.js

```diff
@@ -24,37 +24,65 @@ var _validator = {
     onSuccess: function() {}
   },
 
+  _skipType: function( elem, type ) {
+    var typeAttr = elem.attr( 'type' );
+    var skips = {
+      required:   !elem.prop( 'required' ),
+      email:      typeAttr !== 'email',
+      radiogroup: typeAttr !== 'radio',
+      checkgroup: typeAttr !== 'checkbox'
+    };
+    return skips[type];
+  },
+
   // Validate an individual input for each of the set types
-  // @param   {object} $input The jQuery object of the input
-  // @param   {string} value  The evaluated value of the input
-  // @returns {object}        The status of each of the tested types
-  _validateInput: function( $input, value ) {
+  // @param   {object} elem The jQuery object of the input
+  // @returns {object}      The status of each of the tested types
+  _validateTypes: function( elem ) {
     var status = {};
+    var value = elem.val();
+    var validation = {
+      required: _validate.single( value, { presence: true } ),
+      email:    _validate.single( value, { email: true } )
+    };
 
     $.each( _validator.settings.types, function() {
-      var validation;
       var type = this;
```

Comment:
Instead of 'var type = this' could you just use the arguments passed into the forEach function( key, value )?